src/lib.rs

```diff
@@ -23,6 +42,11 @@ use crate::reconcilable::{Reconcilable, ReconciliationResult};
 
 const BUFFER_SIZE: usize = 65507;
 
+/// The main service of this crate.
+/// Wraps a key-value map
+/// and a collection of peer addresses.
+/// Exposes a run method to run on each instance
+/// to sync.
```

Comment:
I would say that “collection of peer addresses” is an implementation detail. What matters is that it handles reconciliation of the map between two instances over a network:
```suggestion
/// Wraps a key-value map to allow reconciliation between two instances over the network
```

Reply:
Then, maybe:
I would say that “collection of peer addresses” is an implementation detail. What matters is that it handles reconciliation of the map between two instances over a network:
```suggestion
/// Wraps a key-value map to allow reconciliation between two instances over the network. The method `run()` must be called to actually synchronize with peers.
```
And we should probably enforce that `run()` is actually called with the type system (for another PR)

Reply:
I think that the idea that the service _wraps_ a collection of addresses if a bit confusing. What about the following?

```
/// Wraps a key-value map to enable reconciliation between different instances over a network.
///
/// The service also keeps track of the addresses of other instances.
///
/// Provides wrappers for its underlying [Map]'s insertion and deletion methods,
/// as well as its main service method: run.
///
/// This struct does not handle removals. See [RemoveService].
```

Reply:
lgtm